### PR TITLE
INFRT/add LLVM lit to infrt

### PR DESCRIPTION
**Pull request by Superjomn**
### PR types
New features 
### PR changes
Others

### Describe
Add LLVM LIT toolkit to INFRT, making MLIR script easy to test.

### NOTE
1. Disabled Pten dependency for fast compilation and iteration and will restore it later,
2. Rename `infrt_exec` to `infrtexec` to be consistent with the existing `infrtopt` tool,
3. Move `infrt/dialect/mlir_tests` to `infrt/tests/dialect` to make it easy to involve MLIR tests for other modules latters.

### CODE REVIEW

File: tools/infrt/generate_pd_op_dialect_from_paddle_op_maker.py [hunk 1]
```diff
@@ -16,6 +16,10 @@
 from paddle.fluid import core
 from paddle import compat as cpt
 
+ops_having_canonicalization = {"elementwise_add", }
+
+ops_having_canonicalization = {"elementwise_add", }
```
Comment (winter-wang) [inline tools/infrt/generate_pd_op_dialect_from_paddle_op_maker.py]: 这两行代码是不是写重复了？

**Review by winter-wang — DISMISSED** [review]
LGTM

File: tools/infrt/generate_pd_op_dialect_from_paddle_op_maker.py (hunk 1 above)
Comment (Superjomn) [reply tools/infrt/generate_pd_op_dialect_from_paddle_op_maker.py]: 下个PR删下

**Review by winter-wang — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**